client/my-sites/themes/install-theme-button.jsx

```diff
@@ -6,79 +6,82 @@ import { recordTracksEvent } from 'calypso/state/analytics/actions';
 import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
 import isAtomicSite from 'calypso/state/selectors/is-site-automated-transfer';
 import siteHasFeature from 'calypso/state/selectors/site-has-feature';
-import { isJetpackSiteMultiSite, isJetpackSite } from 'calypso/state/sites/selectors';
-import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+import {
+	isJetpackSiteMultiSite,
+	isJetpackSite,
+	getSiteAdminUrl,
+	getSiteSlug,
+} from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 import { trackClick } from './helpers';
 
 import './install-theme-button.scss';
 
+function getInstallThemeUrl( state, siteId ) {
+	if ( ! siteId ) {
+		return '/themes/upload';
+	}
+
+	const atomicSite = isAtomicSite( state, siteId );
+	const siteCanInstallThemes = siteHasFeature( state, siteId, FEATURE_INSTALL_THEMES );
+	if ( atomicSite && siteCanInstallThemes ) {
+		return getSiteAdminUrl( state, siteId, 'theme-install.php' );
+	}
+
+	const siteSlug = getSiteSlug( state, siteId );
```

Comment:
`siteSlug` could return `null`. Should we account for that, since this will result in `themes/upload/null`?

Reply:
I think this won't happen in real life. If `siteId` is `null`, we'll return earlier, before calling `getSiteSlug`. And if `siteId` is valid, the `siteSelection` router middleware will ensure that the site info is loaded in Redux state before it proceeds to rendering the Theme UI.